### DISCUSSION TITLE
${{ github.sha }} giving a completely unique SHA when used in Required Workflows

### DISCUSSION THREAD

**Discussion by ajmalab** [community/community#47804] [Actions] [answered]
### Select Topic Area

Bug

### Body

I am testing out the Required workflows feature that is in public beta. Using `${{ github.sha }}` should ideally be giving me the SHA of the commit that triggered the workflow. This works as expected in Re-usable workflows and normal workflows per repo. However, when used in a Required Workflow, it gives a completely unique (unseen) SHA. I tested it out by comparing both 
`${{ github.event.before }}` and `${{ github.event.after }}` with `${{ github.sha }}`.

**Comment by airtower-luna** [comment, marked as answer]
On what event is your workflow running? For `pull_request` events the `github.sha` will point at a temporary merge commit, basically what would be the result of merging the PR.

**Reply by ajmalab** [reply]
Perhaps this could be it. It's most likely that the temporary merge commit is the one I'm seeing. Thank you!

**Reply by airtower-luna** [reply]
Glad I could help! Please don't forget to mark the answer. :slightly_smiling_face:

If you need to get the SHA of the latest commit on the branch you're trying to merge for a `pull_request` event, `github.event.head.sha` should be the right thing.

**Reply by ajmalab** [reply]
Minor correction, think you mean `github.event.pull_request.head.sha` :)
